esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -36,6 +36,20 @@
 #include "build-config.h"
 
 #include "loggingagentbase.hpp"
+
+bool ensureSDSPath(const char * sdsPath)
+{
+    if (!sdsPath)
+        return false;
+
+    Owned<IRemoteConnection> conn = querySDS().connect(sdsPath, myProcessSession(), RTM_CREATE_QUERY, SDS_LOCK_TIMEOUT_DESDL);
```

Comment:
Should use a write lock explicitly really, i.e.: RTM_LOCK_WRITE|RTM_CREATE_QUERY